coda_v2_python_client/firebase_client_wrapper.py

```diff
@@ -622,3 +622,47 @@ def set_user_ids(self, dataset_id, user_ids):
                 batch.set(self.get_segment_ref(segment_id), {"users": user_ids})
         batch.commit()
         log.debug(f"Wrote {len(user_ids)} users to dataset {dataset_id}")
+
+    MAX_SEGMENT_SIZE = 2500
+
+    def add_message_to_dataset(self, dataset_id, message, max_segment_size=MAX_SEGMENT_SIZE):
+        """
+        Adds message to a given dataset.
+
+        :param dataset_id: Id of the dataset to add the message into.
+        :type dataset_id: str
+        :param message: The message to be added.
+        :type message: core_data_modules.data_models.message.Message
+        :param max_segment_size: the maximum size for a segment, defaults to 2500
+        :type max_segment_size: int, optional
+        """
+        latest_segment_id = self.id_for_segment(dataset_id, self.get_segment_count(dataset_id))
+
+        # Note: We need to read the latest segment so we know how big it is, but there's no need to check previous segments.
+        segment_messages_metrics = self.get_segment_messages_metrics(latest_segment_id)
+        if segment_messages_metrics is None:
+            segment_messages_metrics = self.compute_segment_messages_metrics(latest_segment_id)
+
+        latest_segment_size = segment_messages_metrics.messages_count
+        if latest_segment_size >= max_segment_size:
+            self.create_next_segment(dataset_id)
+            latest_segment_id = self.id_for_segment(dataset_id, self.get_segment_count(dataset_id))
+            segment_messages_metrics = self.compute_segment_messages_metrics(latest_segment_id)
+
+        batch = self._client.batch()
+
+        message = message.copy()
+        message.last_updated = firestore.firestore.SERVER_TIMESTAMP
+
+        message_ref = self.get_message_ref(latest_segment_id, message.message_id)
+        message_snapshot = message_ref.get()
+
+        assert not message_snapshot.exists, f"message with id {message.message_id} already exists."
```

Comment:
This only checks if the message exists in this segment. We need to check if it exists in all segments

Reply:
I'll update in a new PR

Reply:
#22